### ISSUE TITLE
Print method does not include background color while printBackground option set true

### ISSUE THREAD

**Issue by bookspirit** [electron/electron#20583]
### Question
 We are mean to directly call printer to print our pages,  so we found the `BrowserWindow.webContents.print([options], [callback])` method, and with the following options:`{silent: true, printBackground: true}`, 
everything work perfect except that all element's background color(specified in css file) does not come out.

### Issue Details

* **Electron Version:**
  * 5.0.5 5.0.11 and 6.0.12
* **Operating System:**
  * Windows 10
* **Last Known Working Electron version:**
  * None of the versions we have tested

### Expected Behavior
Correctly show the background color of page in PDF file or printed pagers that come out by calling `BrowserWindow.webContents.print([options], [callback])` method.

### Actual Behavior
The background color of all element in the page is not rendered(white color background with border, just act like chrome with include background checked off

### To Reproduce
with the following code
```
let winPDF = new BrowserWindow({
  show: false,
})
winPDF.webContents.loadURL('https://electronjs.org/')
winPDF.on('ready-to-show', () => {
  winPDF.webContents.print(
    {
      silent: true,
      printBackground: true,
    },
    success => {
      winPDF.close()
    }
  )
})
```
### Screenshots
Snapshot of page that has background color
![page](https://user-images.githubusercontent.com/14139170/66828311-49d59880-ef83-11e9-8879-5adb53a9b9f4.png)

Snapshot of print result which print the test page  using the code above.
![print](https://user-images.githubusercontent.com/14139170/66828324-50641000-ef83-11e9-80cc-3d204165d6ab.png)

**Comment by cxyao** [comment]
We also meet the same problem under v5.0.11
webContents.print({ printBackground: true } )not work.

**Comment by barbalex** [comment]
Same here with v7.0.0

**Comment by travenec** [comment]
Confirming the same behavior with v6.0.2.

**Comment by kildem** [comment]
the same problem with v8.0.2

**Comment by barbalex** [comment]
There is a way to achieve this using css:

```css
/* get background colors to show */
@media print {
  -webkit-print-color-adjust: exact;
}
```

**Comment by travenec** [comment]
I am happy to confirm that the CSS property `-webkit-print-color-adjust: exact` @barbalex mentioned solves the problem. (At least with Electron@7.)

**Comment by cxyao** [comment]
@barbalex @travenec Thanks a lot. But `-webkit-print-color-adjust: exact` only work with 7.x and 8.x
Unfortunately Electron 5 and 6 still haven't any solution to slove the problem

**Comment by fabiospampinato** [comment]
This is still a problem on Electron v8.2.2.

**Comment by staifan** [comment]
@fabiospampinato I just encountered the problem today in version 8.2.5 and it finally worked.
I had to put "-webkit-print-color-adjust: exact" in the style of the body

`<body style="min-width: 21cm;-webkit-print-color-adjust: exact">`

and we must also be careful if we use style sheets like bootstrap because they block the operation with media print

> @media print {
  *,
  *:before,
  *:after {
    **color: #000 !important;**
    text-shadow: none !important;
    **background: transparent !important;**
    -webkit-box-shadow: none !important;
    box-shadow: none !important;
  }

you have to think about removing the "color" and "background" parameters

**Comment by DRocksCoding** [comment]
> @media print {
> *,
> *:before,
> *:after {
> **color: #000 !important;**
> text-shadow: none !important;
> **background: transparent !important;**
> -webkit-box-shadow: none !important;
> box-shadow: none !important;
> }

Thanks for this it was really helpful!! (and it works in 8.2.3)

**Comment by jjeff** [comment]
**TLDR;**
``` javascript
  mainWindow.webContents.print({
    printBackground: true // this doesn't work
  })
```
--- Longer version ---
Still a bug in 11.0.0-beta.23. Here's an Electron Fiddle to test:
https://gist.github.com/5f79fd724f205a54f79474304dec5ec8

You don't need to actually print to see the bug. If you're on a Mac and choose either "Open in Preview" or "Save As PDF…" from the "PDF" menu in the corner of the Print dialog, you can see the problem in the resulting print preview PDF. I haven't tested  with `silent` option enabled, but from the rest of this thread, it sounds like that's a problem too.

**Comment by electron-triage** [comment]
The Electron version reported on this issue is no longer supported. See our [supported versions documentation](https://www.electronjs.org/docs/tutorial/support#supported-versions).

If this is still reproducible on a supported version, please open a new issue with any other new information that a maintainer should know.

Thank you for taking the time to report this issue and helping to make Electron better! Your help is appreciated.
